docs/changelog/3_x.rst

```diff
@@ -15,14 +15,57 @@ To play with the new features, install the CLI using
 
   $ edgedb project init --server-version=testing
 
+.. note::
+
+    Good news, everyone! Upgrades from the betas to release candidates (RCs)
+    and to the final release will *not* require a dump and restore. 🎉 You can try
+    out the beta with the assurance that this upgrade to the beta will be the
+    last dump and restore that is required of you.
```

Comment:
... for 3.0 :P

Reply:
Circled back to make it extra clear we're only talking about 3.0 here.